libsnark/zk_proof_systems/plonk/tests/bls12_381_test_vector_transcript_hasher.cpp

```diff
@@ -0,0 +1,138 @@
+/** @file
+ *****************************************************************************
+ * @author     This file is part of libff, developed by Clearmatics Ltd
+ *             (originally developed by SCIPR Lab) and contributors
+ *             (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#ifndef LIBSNARK_ZK_PROOF_SYSTEMS_PLONK_TESTS_BLS12_381_TEST_VECTOR_TRANSCRIPT_HASHER_CPP_
+#define LIBSNARK_ZK_PROOF_SYSTEMS_PLONK_TESTS_BLS12_381_TEST_VECTOR_TRANSCRIPT_HASHER_CPP_
+
+// Implementation of the transcript hasher interface specific to the BLS12-381
+// curve. See bls12_381_test_vector_transcript_hasher.hpp
+namespace libsnark
+{
+
+bls12_381_test_vector_transcript_hasher::
+    bls12_381_test_vector_transcript_hasher()
+    : length_array({288, 320, 416, 704, 896, 1120})
+    , challenge_array({"beta", "gamma", "alpha", "zeta", "nu", "u"})
+{
+    plonk_example example;
```

Comment:
Let's assert that the two arrays are of the same length.

Reply:
Done.